crates/snapbox/src/data.rs

```diff
@@ -102,6 +117,8 @@ impl Data {
         match self.inner {
             DataInner::Binary(data) => Self::binary(data),
             DataInner::Text(data) => Self::text(op(&data)),
+            #[cfg(feature = "json")]
+            DataInner::Json(json) => Self::json(json),
```

Comment:
A lot of this API is centered around text operations.  It makes me wonder if we should keep structured data in here or use a different type for it.

Reply:
I could go either way but it might make the API a bit odd. What do you think the separate type should look like? I would hesitate to suggest a supertype similar to below.
```rust
pub enum SuperDataInner {
    Structured(Structured),
    Text(Text),
}
```

I just don't know enough about how the API would change to give a good suggestion either way